apps/mobile/src/components/amount-field/amount-field.utils.ts

```diff
@@ -0,0 +1,77 @@
+import { decimalSeparator } from '@/features/send-refactor/temporary-constants';
+import { type CurrencyMode } from '@/features/send-refactor/types';
+
+import { currencyDecimalsMap } from '@leather.io/constants';
+import { type Currency, type MarketData, type Money } from '@leather.io/models';
+import { baseCurrencyAmountInQuote, createMoneyFromDecimal } from '@leather.io/utils';
+
+interface FormatPrimaryValueParams {
+  value: string;
+  currency: Currency;
+  showCurrency: boolean;
+  locale: string;
+}
+
+export function formatPrimaryValue({
```

Comment:
It seems like `formatPrimaryValue` and `formatSecondaryValue` could be the same function with a little refactoring?

Reply:
There are some significant differences between the two, the primary value is a live input so it should allow trailing zeros, and only shows currency symbol for crypto entry. I'd argue that these should be always separate, but `formatPrimaryValue` is poorly named, hence the confusion. Will revisit.